js.go

```diff
@@ -984,6 +981,11 @@ func (js *js) subscribe(subj, queue string, cb MsgHandler, ch chan *Msg, isSync
 		if err != nil {
 			return nil, err
 		}
+
+		// With flow control enabled async subscriptions we will set a larger pending bytes and msgs limit than the default.
+		if cb != nil && hasFC {
+			sub.SetPendingLimits(DefaultSubPendingMsgsLimit*4, DefaultSubPendingBytesLimit*4)
```

Comment:
We can match bytes with the FC internals plus a bit of a fudge factor.

I think it is 1M can be outstanding without an FC ack.